### PR TITLE
Antenna deploy: use both I2C lines

### PR DESCRIPTION
**Pull request by Makiv1**
implements #144 

For testing, each command was run on both buses and I verified that the correct led was lit on the electrical model.

### CODE REVIEW

**Comment by DeflateAwning** [comment]
I saw the "don't review yet" tag. What was remaining on this one? I thought it was tested and pretty close to finished?

**Comment by Makiv1** [comment]
Hey Parker, everything was tested and works but I don't know if I'm happy with the quality of code I wrote. Let me look it over today before you start reviewing so you don't waste your time :)

**Comment by Makiv1** [comment]
Should be good for a review now!

File: firmware/Core/Inc/antenna_deploy_drivers/ant_commands.h [hunk 1]
```diff
@@ -22,18 +22,18 @@ struct Antenna_deployment_status {
...
 
-uint8_t ANT_CMD_reset();
-uint8_t ANT_CMD_arm_antenna_system();
-uint8_t ANT_CMD_disarm_antenna_system();
-uint8_t ANT_CMD_deploy_antenna(uint8_t antenna, uint8_t activation_time_seconds);
-uint8_t ANT_CMD_start_automated_sequential_deployment(uint8_t activation_time_seconds);
-uint8_t ANT_CMD_deploy_antenna_with_override(uint8_t antenna, uint8_t activation_time_seconds);
-uint8_t ANT_CMD_cancel_deployment_system_activation();
-uint8_t ANT_CMD_measure_temp(uint16_t *result);
+uint8_t ANT_CMD_reset(enum Ant_i2c_bus_mcu i2c_bus_mcu);
```
Comment (DeflateAwning) [inline firmware/Core/Inc/antenna_deploy_drivers/ant_commands.h:25]: Please rename the enum to be `ANT_i2c_bus_mcu` (capital "ANT")

File: firmware/Core/Inc/antenna_deploy_drivers/ant_commands.h [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #define __INCLUDE_GUARD_ANT_COMMANDS_H__
 
 #include <stdint.h>
-
+#include "antenna_deploy_drivers/ant_internal_drivers.h"
 
 struct Antenna_deployment_status {
```
Comment (DeflateAwning) [inline firmware/Core/Inc/antenna_deploy_drivers/ant_commands.h:7]: Please rename struct to be `ANT_antenna_deployment_status`

File: firmware/Core/Src/antenna_deploy_drivers/ant_commands.c [hunk 3]
```diff
@@ -95,34 +100,41 @@ uint8_t ANT_CMD_deploy_antenna(uint8_t antenna, uint8_t activation_time_seconds)
             cmd_buf[0] = ANT_CMD_DEPLOY_ANTENNA4;
             break;
         default:
-            DEBUG_uart_print_str("Invalid choice for antenna: antenna must be between 1-4 inclusive.");
+            LOG_message(
+                LOG_SYSTEM_ANTENNA_DEPLOY,
+                LOG_SEVERITY_WARNING,
```
Comment (DeflateAwning) [inline firmware/Core/Src/antenna_deploy_drivers/ant_commands.c:105]: Let's call it an "error" if it results in an early return from a function.

File: firmware/Core/Src/antenna_deploy_drivers/ant_internal_drivers.c [hunk 4]
```diff
@@ -1,51 +1,92 @@
...
+ * the deployment of the antennas. It also has two seperate i2c buses which can be used to communicate with the ADS. The combination of 
+ * microcontroller (mcu) to communicate to and i2c line to use is specified using a value from the Ant_i2c_bus_mcu enum. 
+ * 
+ */
+ 
+
+const uint16_t ANT_ADDR_A = 0x31 << 1; // i2c address of mcu A on ant deploy system
+const uint16_t ANT_ADDR_B = 0x32 << 1; // i2c address of mcu B on ant deploy system
+
 const uint8_t timeout = 50;
```
Comment (DeflateAwning) [inline firmware/Core/Src/antenna_deploy_drivers/ant_internal_drivers.c:23]: What units is this? Perhaps also include a comment with info about what the timeout is for (HAL_transmit, HAL_receive? Both?)

File: firmware/Core/Src/antenna_deploy_drivers/ant_internal_drivers.c [hunk 5]
```diff
@@ -1,51 +1,92 @@
...
+    HAL_StatusTypeDef transmit_status;
+    switch(i2c_bus_mcu) {
+        case ANT_I2C_BUS_A_MCU_A:
+            transmit_status = HAL_I2C_Master_Transmit(&hi2c2, ANT_ADDR_A, cmd_buf, cmd_len, timeout);
+            break;
+        case ANT_I2C_BUS_B_MCU_B:
+            transmit_status = HAL_I2C_Master_Transmit(&hi2c3, ANT_ADDR_B, cmd_buf, cmd_len, timeout);
+            break;
+        default:
+            DEBUG_uart_print_str("invalid i2c bus passed");
```
Comment (DeflateAwning) [inline firmware/Core/Src/antenna_deploy_drivers/ant_internal_drivers.c:45]: Please remove the uart print if using LOG_message.

File: firmware/Core/Src/telecommands/antenna_telecommand_defs.c [hunk 6]
```diff
@@ -1,22 +1,45 @@
...
+    case 'B':
+        i2c_bus_mcu = ANT_I2C_BUS_B_MCU_B;
+        break; 
+    default:
+        LOG_message(LOG_SYSTEM_TELECOMMAND,LOG_SEVERITY_WARNING,LOG_SINK_ALL, "Invalid choice for i2c bus");
+        return 1;
+        break;
+    }
+
+    const int status = ANT_CMD_reset(i2c_bus_mcu);
```
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/antenna_telecommand_defs.c:42]: Avoid `int` type

**Review by DeflateAwning — CHANGES_REQUESTED** [review]
Looks awesome! Just a few easy small tweaks

**Review by DeflateAwning — APPROVED** [review]
Thanks!

**Pull request merged**